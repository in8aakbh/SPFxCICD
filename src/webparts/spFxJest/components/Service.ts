import { IListItem } from './IListItem';
import { sp, Items, spODataEntityArray } from "@pnp/sp";


export class Services {
    public getItems(): Promise<IListItem[]> {
        return new Promise<IListItem[]>((resolve, reject) => {
          
            sp.web.lists.getByTitle("Site Pages").items.select("Id", "Title").get(spODataEntityArray<Items, IListItem>(Items)).then(items => {
                resolve(items);
            }).catch(error => reject(error));
        });
    }
}
