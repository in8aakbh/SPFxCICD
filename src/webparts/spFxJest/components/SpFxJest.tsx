import * as React from 'react';
import styles from './SpFxJest.module.scss';
import { ISpFxJestProps } from './ISpFxJestProps';


export default class SpFxJest extends React.Component<ISpFxJestProps, {}> {
  public render(): React.ReactElement<ISpFxJestProps> {
    return (
      <div className={styles.spFxJest}>
      <div className={styles.container}>
        <div className={styles.row}><span><u><b>List Item(s)</b></u></span></div>
        <div className={styles.row}>
          <div className={styles.column}>
            <ul>
              {this.props.ListItem.map((item, i) => { return <li key={item.Title}>{item.Title}</li>; })}
            </ul>
          </div>
        </div>
      </div>
    </div>
    );
  }
}
