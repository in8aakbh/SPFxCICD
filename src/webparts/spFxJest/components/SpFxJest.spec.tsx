import * as React from 'react';
import 'jest';
import { shallow } from 'enzyme';
import { Services } from './Service';
import SpFxJest from './SpFxJest';
import { sp } from "@pnp/sp";
import { SPFetchClient } from "@pnp/nodejs";
let wrapper: any;
beforeEach(() => {
    sp.setup({
        sp: {
            fetchClientFactory: () => {
                return new SPFetchClient("<<Site URL>>", "<<Client ID>>", "<<Client Secret>>");
            },
        }
    });
});
test('getItems() returns correct result', (done) => {
/*     let _Services = new Services();
    _Services.getItems().then(data => {
        wrapper = shallow(<SpFxJest ListItem={data} />); */
        wrapper = shallow(<SpFxJest ListItem={[]} />);
        expect(wrapper.find('span').first().text()).toBe('List Item(s)');
        //console.log(data.length);
        //(data.length > 0) ? expect(wrapper.find('li').length).toBeGreaterThan(2) : expect(wrapper.find('li').length).toBeFalsy();
        //expect(wrapper).toMatchSnapshot();
        //expect(wrapper).toMatchSnapshot();
        done();
    /* }); */
}, 100000);