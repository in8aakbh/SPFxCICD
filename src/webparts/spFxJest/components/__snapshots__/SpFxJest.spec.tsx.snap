// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getItems() returns correct result 1`] = `
<div>
  <div>
    <div>
      <span>
        <u>
          <b>
            List Item(s)
          </b>
        </u>
      </span>
    </div>
    <div>
      <div>
        <ul>
          <li
            key="Sn.No1"
          >
            Sn.No1
          </li>
          <li
            key="Sn.No2"
          >
            Sn.No2
          </li>
          <li
            key="Sn.No3"
          >
            Sn.No3
          </li>
          <li
            key="Sn.No4"
          >
            Sn.No4
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
